iceoryx_posh/test/integrationtests/test_posh_experimental_node.cpp

```diff
@@ -0,0 +1,410 @@
+    auto node2 = std::move(node2_result.value());
+
+    IOX_TESTING_ASSERT_NO_PANIC();
+}
+
+TEST(Node_test, NodeDoesNotUseTheStaticRuntime)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9408ea42-38ab-4547-b7b3-ec2dda2501ba");
+
+    RouDiEnv roudi;
+
+    auto node1 = RouDiEnvNodeBuilder("foo").create().expect("Creating a node should not fail!");
+    auto node2 = RouDiEnvNodeBuilder("bar").create().expect("Creating a node should not fail!");
+
+    EXPECT_THAT(roudi.numberOfActiveRuntimeTestInterfaces(), Eq(0));
+}
+
+TEST(Node_test, CreatingNodeWithInvalidNameLeadsToError)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7a460f65-2970-489f-98e3-2c402fb05766");
+
+    RouDiEnv roudi;
+
+    RouDiEnvNodeBuilder("")
+        .create()
+        .and_then([](const auto&) { GTEST_FAIL() << "Creating a 'Node' with empty name should fail"; })
+        .or_else([](const auto error) { EXPECT_THAT(error, Eq(NodeBuilderError::IPC_CHANNEL_CREATION_FAILED)); });
+
+    RouDiEnvNodeBuilder("/foo")
+        .create()
+        .and_then([](const auto&) { GTEST_FAIL() << "Creating a 'Node' with '/' in name should fail"; })
+        .or_else([](const auto error) { EXPECT_THAT(error, Eq(NodeBuilderError::IPC_CHANNEL_CREATION_FAILED)); });
+}
+
+TEST(Node_test, ReRegisteringNodeWithRunningRouDiWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2ce9d5f0-6989-4302-92b7-458fe1412111");
+
+    RouDiEnv roudi;
+
+    optional<Node> node;
+
+    auto node_result = RouDiEnvNodeBuilder("foo").create();
+    ASSERT_FALSE(node_result.has_error());
+
+    node.emplace(std::move(node_result.value()));
+    node.reset();
+
+    node_result = RouDiEnvNodeBuilder("foo").create();
+    ASSERT_FALSE(node_result.has_error());
+
+    node.emplace(std::move(node_result.value()));
+
+    IOX_TESTING_ASSERT_NO_PANIC();
+}
+
+TEST(Node_test, RegisteringNodeWithoutRunningRouDiWithZeroWaitTimeResultsInImmediateTimeout)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f2041773-84d9-4c9b-9309-996af83d6ff0");
+
+    deadline_timer timer{20_ms};
+
+    auto node_result = RouDiEnvNodeBuilder("foo").create();
+
+    EXPECT_FALSE(timer.hasExpired());
+
+    ASSERT_TRUE(node_result.has_error());
+    EXPECT_THAT(node_result.error(), Eq(NodeBuilderError::TIMEOUT));
+}
+
+TEST(Node_test, RegisteringNodeWithoutRunningRouDiWithSomeWaitTimeResultsInTimeout)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ac069a39-6cdc-4f2e-8b88-984a7d1a5487");
+
+    units::Duration wait_for_roudi_test_timeout{100_ms};
+    units::Duration wait_for_roudi_timeout{2 * wait_for_roudi_test_timeout};
+    deadline_timer timer{wait_for_roudi_test_timeout};
+
+    auto node_result = RouDiEnvNodeBuilder("foo").roudi_registration_timeout(wait_for_roudi_timeout).create();
+
+    EXPECT_TRUE(timer.hasExpired());
+
+    ASSERT_TRUE(node_result.has_error());
+    EXPECT_THAT(node_result.error(), Eq(NodeBuilderError::TIMEOUT));
+}
+
+TEST(Node_test, RegisteringNodeWithDelayedRouDiStartWorks)
```

Comment:
Please also test to create the maximum supported amount of nodes.

Reply:
See https://github.com/eclipse-iceoryx/iceoryx/pull/2208/files#r1506312756